aten/src/ATen/mps/MPSAllocator.mm

```diff
@@ -202,6 +212,9 @@
     block_found =
         // Attempt allocate
         alloc_buffer(params) ||
+        // Callbacks might release more memory (eg. by forcing a GC in the host language) thus
+        // we can retry getting a free buffer in the pool, before trying to alloc again.
+        (trigger_callbacks() && get_free_buffer(params)) ||
```

Comment:
Also, if you're intention was to know "when `get_free_buffer()` fails (due to buffers not available in pool), only then send a callback", then your order of function calling is wrong. You should change it to this instead, otherwise it'll [short-circuit](https://en.wikipedia.org/wiki/Short-circuit_evaluation):
`(get_free_buffer(params) && trigger_callbacks())`